src/Form/DataTransformer/ModelToIdPropertyTransformer.php

```diff
@@ -136,24 +139,23 @@ public function reverseTransform($value)
      * @psalm-return array{_labels?: array<string>}&array<int|string>
      * @phpstan-return array<int|string|array<string>>
      */
-    public function transform($value)
+    public function transform($value): array
     {
         $result = [];
 
         if (!$value) {
             return $result;
         }
 
-        $isArray = \is_array($value);
```

Comment:
I don't know what is the reason for this change. This variable seemed correct given the number of `is_array()` calls checking against its value.
Could you please explain?

Reply:
phpstan or/and psalm doesn't understand correctly if this check is saved in a variable

Reply:
That's sad. Fortunately, this function is optimized by the compiler, but we should take care of cases where that could not be the same.

Reply:
Yes, but since it's optimized by the compiler we don't need for another solution here.

Reply:
I understand. I mean similar cases where the variable could be holding a value returned by a non optimized function.